app/services/facebook/event_repository.rb

```diff
@@ -0,0 +1,35 @@
+class Facebook::EventRepository
+  def load!(event)
+    if cached?(event)
+      FetchFacebookEventWorker.perform_async(event.id)
+      Facebook::Event.new(cache(event))
+    else
+      event.load!
+      cache!(event)
+      event
+    end
+  end
```

Comment:
We can inprove the method, moving the actions to two methods to call if cached or not. Do you think is a good solution?